src/components/timecard/edit-shift-timecard/EditShiftTimecard.jsx

```diff
@@ -101,7 +102,10 @@ const EditShiftTimecard = ({
               `${formatTime(timeEntry.startTime)} to ${
                 timeEntry.finishTime ? formatTime(timeEntry.finishTime) : '-'
               } ${
-                timeEntry.finishNextDay
+                dayjs(timeEntry.finishTime).isAfter(
+                  dayjs(timeEntry.startTime),
+                  'day'
+                )
                   ? ` on ${formatDateNoYear(timeEntry.finishTime)}`
                   : ''
```

Comment:
The jsx is getting a bit confusing to read now with all the additional conditional logic. 
Might be worth taking a quick look to see if some of these bits of logic can be pulled out into tiny functions and then call here to make it a bit more readable/clean.
E.g. `timeEntry.finishTime ? formatTime(timeEntry.finishTime)`
`dayjs(timeEntry.finishTime).isAfter(
                  dayjs(timeEntry.startTime),
                  'day'
                )`

If it ends up creating more overhead feel free to just resolve the comment.

Reply:
This will add a little too much overhead for now, we're looking to fully change how finishNextDay is handled on another ticket so this will be picked up then